### PR TITLE
Add logical_and and logical_or

### PR DESCRIPTION
**Pull request by xuhdev**
With the CI failure caused in 8bbafa0b32d2899ef6101172d62c6049427c977b fixed (incorrect return type of the lambdas in CUDA kernels)

### CODE REVIEW

**Comment by xuhdev** [comment]
@zou3519 Thanks for catching the bug. I have verified locally that the bug caused in 8bbafa0b32d2899ef6101172d62c6049427c977b has been fixed by correcting the return type of the lambdas in CUDA kernels to `bool`.

**Comment by zou3519** [comment]
@xuhdev thanks for fixing this so quickly. I don't have context about this change so it would take me a while to read the entire change; @nairbv as the original reviewer could you re-review this?

File: test/test_torch.py [hunk 1]
```diff
@@ -6521,41 +6522,61 @@ def test_logical_not(self, device):
             a.logical_not_()
             self.assertEqual(expected_res, a)
 
-    def test_logical_xor(self, device):
+    def _test_logical(self, device, op, a_, b_, expected_res_):
         for dtype in torch.testing.get_all_dtypes():
-            expected_res = torch.tensor([0, 0, 1, 1], dtype=dtype, device=device)
-            a = torch.tensor([10, 0, 1, 0], dtype=dtype, device=device)
+            if device.startswith('xla') and dtype == torch.half:  # Skip Half on xla.
```
Comment (ailzhang) [inline test/test_torch.py]: Hmmm we shouldn't put 'xla' in the code branches. This is very hard to maintain... It's okay for XLA for fail on Half for now. (Let's skip in pytorch/xla). 
We'll have a more generalized way to let XLA skip half tests in the future.

Comment (xuhdev) [reply test/test_torch.py]: OK, I have removed the relevant branches. Could you merge https://github.com/pytorch/xla/pull/1325 then?

Comment (xuhdev) [reply test/test_torch.py]: Thanks! Rebased -- expecting all tests to pass

**Comment by xuhdev** [comment]
@ailzhang All tests have passed now

**Review by ailzhang — APPROVED** [review]

**Comment by xuhdev** [comment]
Thanks!

**Pull request closed without merging**